File: src/org/newtonpolyhedron/solve/polyinter/PolyIntersectionSolver.java
package org.newtonpolyhedron.solve.polyinter;

import java.util.List;
import java.util.Map;

import org.newtonpolyhedron.entity.vector.FractionVector;
import org.newtonpolyhedron.entity.vector.IntVector;

public interface PolyIntersectionSolver {
	
	/**
	 * Computes a polyhedron intersections.
	 * 
	 * @param polyhedrons
	 *            source polyhedrons
	 * @param dim
	 *            polyhedron dimensions
	 * @return { vector : [ point indices list per polyhedron ] }
	 * @throws Exception
	 */
	public Map <IntVector, List <List <Integer>>> solve(
			List <List <FractionVector>> polyhedrons,
			int dim) throws Exception;
}

File: src/org/newtonpolyhedron/solve/polyinter/PolyIntersectionSolver.scala
package org.newtonpolyhedron.solve.polyinter

import scala.collection.immutable.SortedSet

import org.fs.utils.collection.table.KeyTable
import org.newtonpolyhedron.entity.vector.VectorImports._

trait PolyIntersectionSolver {
  /**
   * Computes a polyhedron intersections.
   *
   * @param polyhedrons
   *            source polyhedrons
   * @param dim
   *            polyhedron dimensions
   * @return { polyIdx, vector -> [ points giving this vector for this poly when intersecting ] }
   */
  def solve(polyhedrons: Seq[IndexedSeq[FracVec]],
            dim: Int): KeyTable[Int, IntVec, SortedSet[Int]]
}